extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/EventBusConsumer.java

```diff
@@ -231,6 +243,58 @@ private static void invoke(BeanInfo bean, MethodInfo method, ResultHandle messag
         }
     }
 
+    /**
+     * If the return type is CompletionStage use:
+     * <code><pre>
+     * cs.whenComplete((whenResult, whenFailure) -> {
+     *  if (failure != null) {
+     *         message.fail(status, whenFailure.getMessage());
+     *  } else {
+     *         message.reply(whenResult);
+     *  }
+     * })
+     * </pre></code>
+     *
+     * @param messageHandle the message variable
+     * @param invoke the bytecode creator
+     * @return the function
+     */
+    private static FunctionCreator generateWhenCompleteHandler(ResultHandle messageHandle, BytecodeCreator invoke) {
+        FunctionCreator handler = invoke.createFunction(BiConsumer.class);
+        BytecodeCreator bytecode = handler.getBytecode();
+
+        // This avoid having to check cast in the branches
+        AssignableResultHandle whenResult = bytecode.createVariable(Object.class);
+        bytecode.assign(whenResult, bytecode.getMethodParam(0));
+        AssignableResultHandle whenFailure = bytecode.createVariable(Exception.class);
+        bytecode.assign(whenFailure, bytecode.getMethodParam(1));
+        AssignableResultHandle message = bytecode.createVariable(Message.class);
+        bytecode.assign(message, messageHandle);
+
+        BranchResult ifFailureIfNull = bytecode.ifNull(whenFailure);
+        // failure is not null branch - message.fail(failureStatus, failure.getMessage())
+        // In this branch we use the EXPLICIT FAILURE CODE
+        BytecodeCreator failureIsNotNull = ifFailureIfNull.falseBranch();
+        ResultHandle failureStatus = failureIsNotNull.load(ConsumeEvent.EXPLICIT_FAILURE_CODE);
+        ResultHandle failureMessage = failureIsNotNull
+                .invokeVirtualMethod(THROWABLE_GET_MESSAGE, whenFailure);
+        failureIsNotNull.invokeInterfaceMethod(
+                MESSAGE_FAIL,
+                message,
+                failureStatus,
+                failureMessage);
```

Comment:
Looks like we lose the stacktrace here. That's expected?

Reply:
Yes, it's how the vert.x event bus work. Anyway, it's async, stack traces are not very useful in this case. Note that it could also work remotely.